### ISSUE TITLE
CC: Remove filesystem pass-through

### ISSUE THREAD

**Issue by fitzthum** [kata-containers/kata-containers#5310]
Confidential Containers pulls the container images from inside the guest and yet we still have filesystem pass-through enabled. With SEV we usually use 9pfs. Leaving this enabled is a potential security risk. We should remove. We should also try to figure out a way for the guest to reject filesystem pass-through if a malicious host tires to set it up.

CC: @jpecholt I think you've pointed out some of the security issues with 9p in the past.

**Comment by jimcadden** [comment]
I think we'll want to disable fs pass-through for every `guestProtection != noneProtection` (e.g., `pef`, `se`, `sev`, `snp`, `tdx`) Similarly, we'll want to disable block device hotplug support.

**Comment by fitzthum** [comment]
Yeah interesting question about when we should disable these settings. I'm not sure that everyone will want such stringent settings, but it would be nice to base it on some existing flag rather than adding yet another one. 

I have been taking a look at this periodically over the last week and it turns out that it's fairly complicated to not break Kata when you turn off the shared filesystem. We definitely need this for SEV, though.

**Comment by jimcadden** [comment]
> I have been taking a look at this periodically over the last week and it turns out that it's fairly complicated to not break Kata when you turn off the shared filesystem. We definitely need this for SEV, though.

Likewise.
